### PR TITLE
Cluster deployments

### PR DESCRIPTION
**Pull request by jtomasek**

### CODE REVIEW

File: src/components/cluster-deployments/ClusterDeploymentsListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}) => {
+  const {
+    metadata: { uid, name, namespace },
+  } = clusterDeployment;
+  return (
+    <TableRow id={uid} index={index} trKey={uid} style={style}>
+      <TableData>
+        <Link to={`/k8s/ns/${namespace}/${ClusterDeploymentKind}/${name}`}>{name}</Link>
+      </TableData>
+      <TableData>?</TableData>
```
Comment (mareklibra) [inline src/components/cluster-deployments/ClusterDeploymentsListPage.tsx:48]: Let's use either dash or add at least one available and useful column here

File: src/components/types/k8s/cluster-deployment.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export type ClusterDeploymentK8sResource = K8sResourceCommon & {
+  spec?: {
+    baseDomain: string;
+    clusterInstallRef: ClusterInstallRef;
+    clusterName: string;
+    platform: {
+      agentBareMetal: {
+        agentSelector: any;
```
Comment (mareklibra) [inline src/components/types/k8s/cluster-deployment.ts:17]: I wonder if any of those fields is optional. Can you please confirm? Just to be safe ...

File: src/components/cluster-deployments/ClusterDeploymentDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  });
+
+  if (agentsError) throw new Error(agentsError);
+  if (agentClusterInstallError) throw new Error(agentClusterInstallError);
+  if (!(agentsLoaded && agentClusterInstallLoaded)) return <LoadingState />;
+
+  const cluster = getAICluster(clusterDeployment, agentClusterInstall, agents);
+  return (
+    <div className="co-m-pane__body">
+      <pre style={{ fontSize: 10 }}>{JSON.stringify(clusterDeployment, null, 2)}</pre>
```
Comment (mareklibra) [inline src/components/cluster-deployments/ClusterDeploymentDetails.tsx:71]: Are we going to improve that in a follow-up?

File: src/components/cluster-deployments/ClusterDeploymentsListPage.tsx (hunk 1 above)
Comment (jtomasek) [reply src/components/cluster-deployments/ClusterDeploymentsListPage.tsx]: Done

File: src/components/types/k8s/cluster-deployment.ts (hunk 2 above)
Comment (jtomasek) [reply src/components/types/k8s/cluster-deployment.ts]: I made some of the fields optional, let's update the rest in follow ups as needed

File: src/components/cluster-deployments/ClusterDeploymentDetails.tsx (hunk 3 above)
Comment (jtomasek) [reply src/components/cluster-deployments/ClusterDeploymentDetails.tsx]: Yes, I've commented it out for now, I continue working on this, so I'll remove it in follow up patches

**Review by mareklibra — APPROVED** [review]

**Pull request merged**